narwhal/primary/src/block_remover.rs

```diff
@@ -1,196 +1,36 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 use crate::{utils, PayloadToken};
-use config::{Committee, SharedWorkerCache, WorkerId};
+use anyhow::Result;
+use config::{SharedWorkerCache, WorkerId};
 use consensus::dag::{Dag, ValidatorDagError};
 use crypto::PublicKey;
-use fastcrypto::{Digest, Hash};
-use futures::{
-    future::{join_all, try_join_all, BoxFuture},
-    stream::{futures_unordered::FuturesUnordered, StreamExt as _},
-    FutureExt,
-};
+use fastcrypto::Hash;
+use futures::future::try_join_all;
 use itertools::Either;
-use network::{P2pNetwork, UnreliableNetwork};
-use std::{collections::HashMap, sync::Arc, time::Duration};
+use network::{P2pNetwork, PrimaryToWorkerRpc};
+use std::{collections::HashMap, sync::Arc};
 use storage::CertificateStore;
 use store::{rocks::TypedStoreError, Store};
-use tokio::{
-    sync::{mpsc, oneshot, watch},
-    task::JoinHandle,
-    time::timeout,
-};
-use tracing::{debug, error, info, instrument, warn};
+
+use tracing::{debug, warn};
 use types::{
-    metered_channel::{Receiver, Sender},
-    BatchDigest, BlockRemoverError, BlockRemoverErrorKind, BlockRemoverResult, Certificate,
-    CertificateDigest, Header, HeaderDigest, PrimaryWorkerMessage, ReconfigureNotification,
+    metered_channel::Sender, BatchDigest, Certificate, CertificateDigest, Header, HeaderDigest,
 };
 
-const BATCH_DELETE_TIMEOUT: Duration = Duration::from_secs(2);
-
-type RequestKey = Vec<u8>;
-
 #[cfg(test)]
 #[path = "tests/block_remover_tests.rs"]
 pub mod block_remover_tests;
 
-#[derive(Debug)]
-pub enum BlockRemoverCommand {
-    RemoveBlocks {
-        // the block ids to remove
-        ids: Vec<CertificateDigest>,
-        // the channel to communicate the results
-        sender: mpsc::Sender<BlockRemoverResult<RemoveBlocksResponse>>,
-    },
-}
-
-#[derive(Clone, Debug)]
-pub struct RemoveBlocksResponse {
-    // the block ids removed
-    ids: Vec<CertificateDigest>,
-}
-
-pub type DeleteBatchResult = Result<DeleteBatchMessage, DeleteBatchMessage>;
-
-#[derive(Clone, Default, Debug)]
-pub struct DeleteBatchMessage {
-    pub ids: Vec<BatchDigest>,
-}
-
 /// BlockRemover is responsible for removing blocks identified by
 /// their certificate id (digest) from across our system. On high level
 /// It will make sure that the DAG is updated, internal storage where
 /// there certificates and headers are stored, and the corresponding
 /// batches as well.
-///
-/// # Example
```

Comment:
I think it's so simple now that an example isn't as useful as it was before—you just construct one of these things and then call its one and only public function. It's harder than before to get an example working in a self-contained way because now it requires constructing an entire anemo::Network instead of just some fake channels. I think the tests are the best usage example at the moment.